internal/transport/http2_client.go

```diff
@@ -742,15 +742,12 @@ func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (*Stream,
 		endStream: false,
 		initStream: func(id uint32) error {
 			t.mu.Lock()
-			if state := t.state; state != reachable {
+			// we want initStream to cleanup and return an error when transport is closing.
+			// initStream is never called when transport is draining.
+			if t.state == closing {
```

Comment:
`// TODO: handle transport closure in loopy instead and remove this.` e.g.

Reply:
done